### ISSUE TITLE
Wrong "problem in file" reported

### ISSUE THREAD

**Issue by adrian-constantin** [microsoft/vscode-cpptools#2280]
Type: LanguageService
**----- Input information below -----**
Looks like a problem with an extern object declaration that includes the class definition

**Describe the bug**
- OS and Version: Ubuntu 18.04 LTS
- VS Code Version: 1.25.1
- C/C++ Extension Version: 0.17.6
- Other extensions you installed (and if the issue persists after disabling them):
    C/C++ intelliSense 0.2.2
- A clear and concise description of what the bug is.
Extern object declaration in header file marked in red as incompatible with the definition, although the code is correct and works as expected.

**To Reproduce**
*Provide a code sample including configuration files such as c_cpp_properties.json*
Include the bellow header declaration:
extern class ApplicationLoop: public MainLoop
{
protected:
    enum State
    {
        ReceivePublicPoint, ReceiveSaltMaterial, ReceiveSignerCertificate, ReceiveDeviceCertificate, ReceiveChallengeSignature, Secured
    }
        appState = ReceivePublicPoint;
    
    unsigned long send_counter = 0, recv_counter = 0;

    virtual void datagram_received(std::size_t message_size, std::uint8_t const message[]) override;
    virtual void timer_ellapsed() override;

public:
    virtual bool init(enum Role role) override;
}
    mainLoop;

And the bellow definition in the .cpp:
ApplicationLoop mainLoop;

**Expected behavior**
No code lines marked in red, as the source code is correct.

**Screenshots**
![screenshot_20180716_144018](https://user-images.githubusercontent.com/35807909/42756751-3dcaa7ae-8906-11e8-9de7-1d42a4c85d97.png)

**Comment by sean-mcmanus** [comment]
What is the definition of MainLoop?

I used
```
class MainLoop
{
protected:
    virtual void datagram_received(std::size_t message_size, std::uint8_t const message[]);
virtual void timer_ellapsed();

public:
virtual bool init(enum Role role);
};
```

What compilerPath is being used? I'm not able to repro this so far with gcc or clang compilerPath on Linux.

![image](https://user-images.githubusercontent.com/19859882/42782905-2222f0ee-88ff-11e8-9580-b6df1a859338.png)

**Comment by adrian-constantin** [comment]
Sorry, now I can not reproduce the issue either.

I just kept working on my project and I no longer have the exact sources I had when I saw the problem. I also tried to re-create a minimal example, but it works properly with no issue. At the time, I thought any declaration of the form extern class Name { ... } varName; would trigger the error.

So for lack of more info I guess this issue can be closed now, sorry for the trouble and thank you for the effort !
